### PR TITLE
Add unwrap method

### PR DESCRIPTION
**Pull request by overdrivenpotato**
Hello :-) This implements `.unwrap()` on `CollectArrayResult`. I've added tests and made sure they pass through `miri`, as well.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -304,4 +326,56 @@ mod test {
...
+
+        let ok = panic::catch_unwind(|| {
+            vec![DropCounter(drop_count.clone())]
+                .into_iter()
+                .collect::<CollectArrayResult<_, 3>>()
+                .unwrap()
+        });
+        drop(ok);
+
+        assert_eq!(1, Arc::try_unwrap(drop_count).unwrap().into_inner());
```
Comment (illicitonion) [inline src/lib.rs]:
```suggestion
        
        assert!(ok.is_err());

        assert_eq!(1, Arc::try_unwrap(drop_count).unwrap().into_inner());
```

I suspect this was a mis-copy&paste - the drop of the elements here should be happening automatically, but we want to make sure we returned an error.

**Review by illicitonion — COMMENTED** [review]
Thanks for the contribution! It looks great - just one small comment and I'm happy to merge and release!

Out of interest, how did you make sure they pass through `miri`?

**Comment by overdrivenpotato** [comment]
Ah thanks I missed that, applied your suggestion.

I ran them through `miri` with the nightly toolchain, all I did was install the miri component through rustup and then I ran `cargo +nightly miri test`. It runs all the unit tests through miri this way for some extra undefined behaviour checks.

**Comment by illicitonion** [comment]
Thanks!

**Comment by illicitonion** [comment]
And [released](https://crates.io/crates/collect_array/0.1.2)!

**Pull request merged**